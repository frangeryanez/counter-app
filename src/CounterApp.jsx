import { useState } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'react-bootstrap';

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);

  const handleSubtract = () => setCounter(counter - 1);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{ counter }</h2>
      <div xs={12} className="pt-2">
        <Button 
          aria-label="btn-add"
          className="me-2" 
          variant="primary" 
          onClick={ handleAdd }
        >
          +1
        </Button>
        <Button 
          aria-label="btn-subtract"
          className="me-2" 
          variant="danger" 
          onClick={ handleSubtract }
        >
          -1
        </Button>
        <Button 
          aria-label="btn-reset"
          className="me-2" 
          variant="info" 
          onClick={ handleReset }
        >
          Reset
        </Button>
      </div>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired
};

CounterApp.defaultProps = {
  value: 0
};