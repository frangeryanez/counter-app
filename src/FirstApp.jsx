const subtitle = 'Subtitle';

export const FirstApp = ({ greetings }) => {
  return (
    <>
      <h1>{ greetings }</h1>
      <p>{ subtitle }</p>
    </>
  );
};