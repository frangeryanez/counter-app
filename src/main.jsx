import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import './styles.css';
import 'bootstrap/dist/css/bootstrap.min.css';
// import { HelloWorldApp } from './HelloWorldApp';
// import { FirstApp } from './FirstApp';

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <CounterApp />
  </React.StrictMode>
);