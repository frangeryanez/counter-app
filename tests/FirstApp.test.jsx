import { render } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';

describe('Test on file <FirstApp />', () => {
  test('Must match the snapshot', () => {
    const greeting = 'Hello World';
    const { container } = render(<FirstApp greetings={greeting} />);

    expect(container).toMatchSnapshot();
  });
  
  test('Should show the message "Hello World" and display the message in an h1', () => {
    const greeting = 'Hello World';
    const { container, getByText } = render(<FirstApp greetings={greeting} />);
    const h1 = container.querySelector('h1');

    expect(getByText(greeting)).toBeTruthy();
    expect(h1.innerHTML).toContain(greeting);
  });
});