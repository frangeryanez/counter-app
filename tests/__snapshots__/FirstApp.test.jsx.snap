// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on file <FirstApp /> Must match the snapshot 1`] = `
<div>
  <h1>
    Hello World
  </h1>
  <p>
    SubTitle
  </p>
</div>
`;
